components/emoji_picker/components/emoji_picker_item.jsx

```diff
@@ -6,74 +6,91 @@ import React from 'react';
 
 import EmojiStore from 'stores/emoji_store.jsx';
 
+const SCROLLING_ADDT_VISUAL_SPACING = 10; // to make give the emoji some visual 'breathing room'
+
 export default class EmojiPickerItem extends React.PureComponent {
     static propTypes = {
         emoji: PropTypes.object.isRequired,
         onItemOver: PropTypes.func.isRequired,
-        onItemOut: PropTypes.func.isRequired,
         onItemClick: PropTypes.func.isRequired,
-        onItemUnmount: PropTypes.func.isRequired,
-        category: PropTypes.string.isRequired
-    }
+        category: PropTypes.string.isRequired,
+        isSelected: PropTypes.bool,
+        categoryIndex: PropTypes.number.isRequired,
+        emojiIndex: PropTypes.number.isRequired,
+        containerRef: PropTypes.any,
+        containerTop: PropTypes.number.isRequired,
+        containerBottom: PropTypes.number.isRequired
+    };
 
     constructor(props) {
         super(props);
-
         this.handleMouseOver = this.handleMouseOver.bind(this);
-        this.handleMouseOut = this.handleMouseOut.bind(this);
         this.handleClick = this.handleClick.bind(this);
+        this.emojiItemRef = (emojiItem) => {
```

Comment:
You can actually declare this outside of the constructor since it'll technically be a method like
```
constructor(props) {
    ...
}

emojiItemRef = (emojiItem) => {
    this.emojiItem = emojiItem;
};
```

Reply:
I honestly did not realize you can use arrow functions in the class like that.